google-cloud-storage/src/main/java/com/google/cloud/storage/spi/v1/HttpStorageRpc.java

```diff
@@ -244,6 +246,29 @@ public StorageObject create(StorageObject storageObject, final InputStream conte
     }
   }
 
+  @Override
+  public StorageObject create(StorageObject storageObject, final RestartableInputStream content,
```

Comment:
You need to choose on which level retires should be performed: `Storage` or the `StorageRpc`. It seems the retries normally performed on Storage level (and this is what you do calling `runWithRetries()` in `StorageImpl.create(BlobInfo, RestartableInputStream, BlobWriteOption...)` implementation above.The `Storage.create()` calls `StorageRpc.create()`, if you retry (or have retry-like code, which you have here with `RetryableInputStreamContent`) you can potentially have `n^2` retires instead of `n` retries.

Most probably  you do not need neither this method nor `RetryableInputStreamContent` class.

Reply:
The only reason that I created another level of indirection is that it is explicitly said that the abstract class `InputStreamContent` does not support retry per [here](https://cs.corp.google.com/piper///depot/google3/java/com/google/api/client/http/InputStreamContent.java?q=InputStreamContent+lang:java&sq=package:piper+file://depot/google3+-file:google3/experimental&dr=CSs&l=23), so reusing `InputStreamContent` might be causing ambiguity and confusion. If that won't be a problem, I can remove this method and the `RetryableInputStreamContent` class.